packages/optimizer/lib/transformers/PreloadHeroImage.js

```diff
@@ -216,6 +225,35 @@ class PreloadHeroImage {
     }
     return {width: 0, height: 0};
   }
+
+  generateImg(node) {
+    if (!this.experimentImg) {
+      return;
+    }
```

Comment:
I think we were only going to do this transform for `amp-img`s that have a `data-hero` attr set. Since we can't guarantee `loading=lazy` behavior for most browsers yet, we don't want to load a bunch of far-from-viewport images.

Reply:
Introduce a max upper bound of 2 hero images and switched to `loading=eager` for these.